src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_binary.cs

```diff
@@ -46,7 +45,7 @@ public static partial class ReaderExtensions
                 return false;
             }
 
-            value = MemoryMarshal.Read<T>(tempSpan);
+            value = Unsafe.ReadUnaligned<T>(ref MemoryMarshal.GetReference(tempSpan));
```

Comment:
I am not sure if we should be relying on the Unsafe class for performance here, especially for the slow path.

Reply:
We're simply skipping the redundant bounds checks here. While I would prefer not to, I don't know that we have a choice to meet our performance goals for now. Hopefully this sort of thing will be taken care of automagically by the JIT in the future.

Reply:
In this case the "slow" path isn't _that_ slow, so it definitely measures. The amortization we're talking about is our best guess- while we don't expect that we'll hit the boundary often, in this case the change is relatively benign and will definitely help where we've guessed wrong.